bigtable-client-core-parent/bigtable-client-core/src/main/java/com/google/cloud/bigtable/grpc/async/AbstractRetryingOperation.java

```diff
@@ -141,7 +144,10 @@ protected void onError(Status status, Metadata trailers) {
     Code code = status.getCode();
     // CANCELLED
     if (code == Status.Code.CANCELLED) {
-      completionFuture.cancel(true);
+      if (!completionFuture.isDone()) {
+        // cancel(true) would trigger GrpcFuture.interuptTask().  We don't want that to happen.
+        completionFuture.cancel(false);
```

Comment:
Would you mind updating the comment as to why we don't want to interrupt the task? If I understand correctly its because task interruption should only come from the user, and since this is coming from the grpc/server the task is already interrupted?